packages/dart/sshnoports/bundles/shell/install.sh

```diff
@@ -2,382 +2,481 @@
 
 # SYSTEM GIVENS #
 is_root() {
-  [ "$(id -u)" -eq 0 ]
+	[ "$(id -u)" -eq 0 ]
 }
 
+unset binary_dir
 user_home=$HOME
 define_env() {
-  script_dir="$(dirname -- "$( readlink -f -- "$0"; )")"
-  bin_dir="/usr/local/bin"
-  systemd_dir="/etc/systemd/system"
-  if is_root; then
-    user="$SUDO_USER"
-    if [ -z "$user" ]; then
-      user="root"
-    else
-      # we are root, but via sudo
-      # so get home directory of SUDO_USER
-      user_home=$(sudo -u "$user" sh -c 'echo $HOME')
-    fi
-  else
-    user="$USER"
-  fi
-  user_bin_dir="$user_home/.local/bin"
-  user_sshnpd_dir="$user_home/.sshnpd"
-  user_log_dir="$user_sshnpd_dir/logs"
-  user_ssh_dir="$user_home/.ssh"
+	script_dir="$(dirname -- "$(readlink -f -- "$0")")"
+	bin_dir="/usr/local/bin"
+	systemd_dir="/etc/systemd/system"
+	if is_root; then
+		user="$SUDO_USER"
+		if [ -z "$user" ]; then
+			user="root"
+		else
+			# we are root, but via sudo
+			# so get home directory of SUDO_USER
+			user_home=$(sudo -u "$user" sh -c 'echo $HOME')
+		fi
+	else
+		user="$USER"
+	fi
+	launchd_dir="$user_home/Library/LaunchAgents"
+	user_bin_dir="$user_home/.local/bin"
+	user_sshnpd_dir="$user_home/.sshnpd"
+	user_log_dir="$user_sshnpd_dir/logs"
+	user_ssh_dir="$user_home/.ssh"
 }
 
 is_darwin() {
-  [ "$(uname)" = 'Darwin' ]
+	[ "$(uname)" = 'Darwin' ]
+}
+
+sedi() {
+	if is_darwin; then
+		sed -i '' "$@"
+	else
+		sed -i "$@"
+	fi
```

Comment:
because macos sed -i requires an argument after, and gnu sed expects the argument omitted if there's no backup extension...